src/main/java/com/metamx/emitter/core/EmitterBuilder.java

```diff
@@ -51,12 +56,24 @@ public void setLoggingEmitterConfig(LoggingEmitterConfig loggingEmitterConfig)
     this.loggingEmitterConfig = loggingEmitterConfig;
   }
 
+  public ParametrizedUriHttpEmitterConfig getParametrizedUriHttpEmitterConfig()
+  {
+    return parametrizedUriHttpEmitterConfig;
+  }
+
+  public void setParametrizedUriHttpEmitterConfig(ParametrizedUriHttpEmitterConfig parametrizedUriHttpEmitterConfig)
+  {
+    this.parametrizedUriHttpEmitterConfig = parametrizedUriHttpEmitterConfig;
+  }
+
   public Emitter build(ObjectMapper objectMapper, HttpClient httpClient, Lifecycle lifecycle)
   {
     if (loggingEmitterConfig != null) {
       return buildLogging(objectMapper, lifecycle);
     } else if (httpEmitterConfig != null) {
       return buildHttp(httpClient, objectMapper, lifecycle);
+    } else if (parametrizedUriHttpEmitterConfig != null) {
```

Comment:
this is getting silly :-/ can the config just take in a factory json object and allow this to materialize the factory with the ObjectMapper?

Then build the `Emitter` with the factory and return.